File: Challenges/June_2021/128.java
class Solution {
    HashSet<Integer> hash = new HashSet<>(); 
    public int longestConsecutive(int[] nums) {
        
        // add all elements to the hash 
        for (int i = 0 ; i < nums.length; i++){
            hash.add(nums[i]); 
        }
        
        int longestSeq = 0; 
        for (int num: nums){
            int currentSeq = 1; 
            if (hash.contains(num-1) == false){
                while(hash.contains(num+currentSeq)){
                    currentSeq ++; 
                }
                
            }
            longestSeq = Math.max(currentSeq, longestSeq); 
        }
        
        return longestSeq; 
    }
    
}

File: 128.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        HashSet<int> set = new HashSet<int>(); 
        var maxSize = 0; 
        foreach (var num in nums) {
            set.Add(num);
        }

        foreach (var num in nums){
            var target = num; 
            var size =  1; 
            while(set.Contains(target+1)) {
                size += 1; 
                target += 1; 
                set.Remove(target); 
            }
            target = num; 
            while(set.Contains(target-1)){
                size += 1;
                target -= 1;  
                set.Remove(target); 
            }
            maxSize = Math.Max(size, maxSize); 
        }
        return maxSize; 
    }
}
